src/foam/core/stdlib.js

```diff
@@ -568,9 +576,10 @@ foam.LIB({
     function getTime(d) { return ! d ? 0 : d.getTime ? d.getTime() : d ; },
     function equals(a, b) { return this.getTime(a) === this.getTime(b); },
     function compare(a, b) {
+      if ( ! this.isInstance(b) && ! foam.Number.isInstance(b) ) return -1;
```

Comment:
Aren't we assuming that type-specific comparators are of the same type? If so, are these kinds of checks needed?